src/injected-command-handler.ts

```diff
@@ -0,0 +1,456 @@
+      "i"
+    );
+
+    for (let i = 0; i < snapshot.snapshotLength; i++) {
+      const item = snapshot.snapshotItem(i);
+      if (
+        item !== null &&
+        this.inViewport(item as HTMLElement) &&
+        (item as HTMLElement).innerText.match(re)
+      ) {
+        matches.push(item);
+      }
+    }
+
+    return matches;
+  }
+
+  private nodesMatchingSelector(selector: string) {
+    let matches = [];
+    const nodes = document.querySelectorAll(selector);
+    for (let i = 0; i < nodes.length; i++) {
+      const item = nodes[i];
+      if (item !== null && this.inViewport(item as HTMLElement)) {
+        matches.push(item);
+      }
+    }
+    return matches;
+  }
+
+  private elementIsScrollable(element: HTMLElement, direction: string): boolean {
+    if (direction === "up" || direction === "down" || direction === "bottom" || direction === "top") {
+      const overflowStyle = window.getComputedStyle(element).overflowY;
+      return (
+        element.scrollHeight > element.clientHeight && (overflowStyle === "scroll" || overflowStyle === "auto")
+      );
+    } else if (direction === "left" || direction === "right") {
+      const overflowStyle = window.getComputedStyle(element).overflowX;
+      return (
+        element.scrollWidth > element.clientWidth && (overflowStyle === "scroll" || overflowStyle === "auto")
+      );
+    }
+    return false;
+  }
+
+  private scrollOptions(element: HTMLElement | Window, direction: string) {
+    let dir = {};
+    if ("clientWidth" in element) {
+      // Element type
+      switch (direction) {
+        case "left":
+          dir = { left: -element.clientWidth * 0.6 };
+          break;
+        case "right":
+          dir = { left: element.clientWidth * 0.6 };
+          break;
+        case "up":
+          dir = { top: -element.clientHeight * 0.6 };
+          break;
+        case "down":
+          dir = { top: element.clientHeight * 0.6 };
+          break;
+        case "bottom":
+          dir = { top: element.scrollHeight };
+          break;
+        case "top":
+          dir = { top: 0 };
+          break;
+      }
+    } else {
+      // Window type
+      switch (direction) {
+        case "left":
+          dir = { left: -element.innerWidth * 0.8 };
+          break;
+        case "right":
+          dir = { left: element.innerWidth * 0.8 };
+          break;
+        case "up":
+          dir = { top: -element.innerHeight * 0.8 };
+          break;
+        case "down":
+          dir = { top: element.innerHeight * 0.8 };
+          break;
+        case "bottom":
+          dir = { top: document.body.scrollHeight };
+          break;
+        case "top":
+          dir = { top: 0 };
+          break;
+      }
+    }
+    let options = Object.assign(dir, { behavior: "smooth" }) as ScrollOptions;
+    return options;
+  }
+
+  private async scrollInDirection(direction: string) {
+    let hoveredElements = document.querySelectorAll("*:hover");
```

Comment:
not a big deal, but there's a bunch of places where we have `let` that could be `const` instead. just a nice hint to the reader/editor that they don't have to worry about mutability.